retrieval/target.go

```diff
@@ -196,11 +198,14 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	if cfg.BasicAuth != nil {
 		t.url.User = url.UserPassword(cfg.BasicAuth.Username, cfg.BasicAuth.Password)
 	}
+	t.url.RawQuery = cfg.Params.Encode()
```

Comment:
That could work, we'd probably need an extra option to handle full deletion of a parameter at some point.